addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/launch-product/launch-product.js

```diff
@@ -356,6 +363,35 @@ class LaunchProduct extends StepBase {
     }
     return result.LaunchPathSummaries[0];
   }
+
+  /**
+   * Check if the resolved input parameter contains a static namespace param. If so, augments the namespace to begin with 
+   * 'analysis-' for permissions purposes (if it does not already start with that) and to end with a unique datetime string 
+   * so Cloudformation doesn't create duplicate stacks for separate workspaces.
+   * 
+   * @param resolvedInputParams 
+   * @param datetime 
+   * @returns {Promise<{namespaceParam:string, namespaceIndex:string}[]>}
+   */
+  async getNamespaceAndIndexIfNecessary(resolvedInputParams, datetime){
+    const namespaceIndex = resolvedInputParams.findIndex(element => element.Key === 'Namespace');
+    let namespaceParam = resolvedInputParams[namespaceIndex].Value;
+    console.log(namespaceParam);
+
+    // Check to make sure the resolved namespace variable begins with 'analysis-' so our templates will allow it
+    if(!namespaceParam.startsWith('analysis-')){
+      namespaceParam = 'analysis-' + namespaceParam;
+    }
+
+    // Check to make sure the resolved namespace variable ends with a unique datetime string so it will be unique for each deployment of a configuration with a static namespace
+    if(namespaceParam.split('-').pop() !== datetime.toString()){
+      namespaceParam += '-' + datetime;
+    }
+    console.log(datetime);
+
+    console.log(namespaceParam);
+    return [namespaceParam, namespaceIndex];
```

Comment:
Usually it's recommended that you return a dictionary like { namespaceParam, namespaceIndex }. And you can access it by doing something like 

```
const { namespaceParam, namespaceIndex } = await this.getNamespaceAndIndexIfNecessary(resolvedInputParams, datetime);
```

Reply:
So I tried to do this and it did not work. The tests failed and said the things returned were undefined. Do I have to name the things I assign to the returned values the same as I did within the function? I had said

`const { afterNamespace, afterIndex } = await this.getNamespaceAndIndexIfNecessary(...`

and it said that afterNamespace was undefined. Or, is it just good practice to name them the same?

Reply:
Nevermind--got it to work this time...don't know what I was doing differently the first time. I made it a dictionary!